Update stripe.mdx - Explain how to localize PaymentSheet on iOS (#31661)
expo/expo: docs/pages/versions/unversioned/sdk/stripe.mdx

[`Linking.createURL()`](/versions/latest/sdk/linking/#createurloptions) will ensure you're using the proper scheme, whether you're running in Expo Go or your production app. `'/--/'` is necessary in Expo Go because it indicates that the substring after it corresponds to the deep link path, and is not part of the path to the app itself.

#### PaymentSheet localization on iOS

> Translation of PaymentSheet is automatically detected based on device's langage, at least on Android. For this to work on iOS, you must add `CFBundleAllowMixedLocalizations` and `CFBundleLocalizations` declarations in `app.json` (or `app.config.js`, `app.config.ts`). 

Reviewer: ```suggestion
On Android, the translation of `PaymentSheet` is automatically detected based on a device's language settings.

On iOS, you must enable  `CFBundleAllowMixedLocalizations` and add the preferred language using `CFBundleLocalizations` under [`ios.infoPlist`](/versions/latest/config/app/#infoplist) in the app config:
```

> ```

Reviewer: ````suggestion
```app.json
````

> {
>   "expo": {
>     "ios": {
>       "infoPlist": {
>           "CFBundleAllowMixedLocalizations": true,
>           "CFBundleLocalizations": ['fr']
>           ...
>       }
>       ...
>     }
>   }
> }

Reviewer: ```suggestion
{
  "expo": {
    "ios": {
      "infoPlist": {
        "CFBundleAllowMixedLocalizations": true,
        "CFBundleLocalizations": ["fr"]
        /* @hide ... */ /* @end */
      }
      /* @hide ... */ /* @end */
    }
  }
}
```

[`Linking.createURL()`](/versions/latest/sdk/linking/#createurloptions) will ensure you're using the proper scheme, whether you're running in Expo Go or your production app. `'/--/'` is necessary in Expo Go because it indicates that the substring after it corresponds to the deep link path, and is not part of the path to the app itself.

#### PaymentSheet localization on iOS

On Android, the translation of `PaymentSheet` is automatically detected based on a device's language settings.

> On iOS, you must enable  `CFBundleAllowMixedLocalizations` and add the preferred language using `CFBundleLocalizations` under [`ios.infoPlist`](/versions/latest/config/app/#infoplist) in the app config:

Reviewer: ```suggestion
On iOS, you must enable `CFBundleAllowMixedLocalizations` and add the preferred language using `CFBundleLocalizations` under [`ios.infoPlist`](/versions/latest/config/app/#infoplist) in the app config:
```

> ```app.json

Reviewer: ````suggestion
```json app.json
````
